tests/unit/frameworks/vernon/self_supervised_learning_test.py

```diff
@@ -0,0 +1,124 @@
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+
+import unittest
+
+import torch
+import torch.nn as nn
+import torchvision.transforms as transforms
+from torchvision.datasets.fakedata import FakeData
+
+from nupic.research.frameworks.pytorch.self_supervised_utils import EncoderClassifier
+from nupic.research.frameworks.vernon import SelfSupervisedExperiment
+
+
+class AutoEncoder(torch.nn.Module):
+    def __init__(self, input_dim=784, hidden_dim=20):
+        super().__init__()
+        self.fc1 = nn.Linear(input_dim, hidden_dim)
+        self.fc2 = nn.Linear(hidden_dim, input_dim)
+
+    def forward(self, x):
+        encoded = self.encode(x)
+        decoded = self.fc2(encoded).view(-1, 1, 28, 28)
+        return decoded
+
+    def encode(self, x):
+        x = x.flatten(start_dim=1)
+        encoded = self.fc1(x)
+        return encoded
+
+
+class LinearClassifier(torch.nn.Module):
+    def __init__(self, input_dim=20, num_classes=10):
+        super().__init__()
+        self.fc = nn.Linear(input_dim, num_classes)
+
+    def forward(self, x):
+        out = self.fc(x)
+        return out
+
+
+# # MNIST
+# MEAN = 0.13062755
+# STDEV = 0.30810780
+# transform = transforms.Compose(
+#     [transforms.ToTensor(), transforms.Normalize(MEAN, STDEV)]
+# )
+
+
+fake_data_args = dict(
+    size=1000, image_size=(1, 28, 28), num_classes=10, transform=transforms.ToTensor()
+)
+
+self_supervised_config = dict(
+    experiment_class=SelfSupervisedExperiment,
+    num_classes=10,
+    # Dataset
+    dataset_class=FakeData,
+    dataset_args=dict(
+        unsupervised=fake_data_args,
+        supervised=fake_data_args,
+        validation=fake_data_args,
+    ),
+    # Number of epochs
+    epochs=5,
+    epochs_to_validate=[2, 4],
+    supervised_training_epochs_per_validation=1,
+    batch_size=32,
+    batch_size_supervised=32,
+    # Model class. Must inherit from "torch.nn.Module"
+    model_class=AutoEncoder,
+    # model model class arguments passed to the constructor
+    model_args=dict(),
+    classifier_model_class=LinearClassifier,
+    classifier_model_args=dict(),
+    optimizer_class=torch.optim.Adam,
+    # Optimizer class class arguments passed to the constructor
+    optimizer_args=dict(lr=0.001),
+    classifier_optimizer_class=torch.optim.SGD,
+    classifier_optimizer_args=dict(lr=0.001),
+    loss_function=torch.nn.functional.mse_loss,
+    loss_function_supervised=torch.nn.functional.cross_entropy,
+)
+
+
+class SelfSupervisedLearningTest(unittest.TestCase):
+    """
+    This is a test class for the `SelfSupervisedExperiment` class.
+    """
+
+    def test_self_supervised_experiment(self):
+        # Setup experiment and initialize model.
+        exp = self_supervised_config["experiment_class"]()
+        exp.setup_experiment(self_supervised_config)
+        assert isinstance(exp.model, EncoderClassifier)
```

Comment:
In a `unittest.TestCase` class, you should use `self.assertTrue`. This builtin class function handles the asserts more elegantly. There are also other ones like `self.assertEqual`. According to the documentation, "these methods are used instead of the assert statement so the test runner can accumulate all test results and produce a report."

Reply:
Changed to the proper asserts, good catch.